libraries/Microsoft.Bot.Builder/CloudAdapterBase.cs

```diff
@@ -0,0 +1,320 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Security.Principal;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Microsoft.Bot.Builder
+{
+    /// <summary>
+    /// An adapter that implements the Bot Framework Protocol and can be hosted in different cloud environmens both public and private.
+    /// </summary>
+    public class CloudAdapterBase : BotAdapter
```

Comment:
Shouldn't it be abstract? (We only have a protected constructor so I assume this class is meant to be subclassed, BotAdapter is abstract too so I think this one could be ablstract as well)

Reply:
There are no abstract methods so its not abstract - it was but I took that method off - so made the class not abstract.

Making the constructor protected forces subclassing right? Is there any reason to make this abstract (assuming we don't introduce an abstract method.)

Reply:
I'm just not used to see something with protected constructors that is not abstract. Not sure if that's a common approach, I personally haven't seen this done, I explicitly mark classes as abstract  and add a protected constructor if I need one but not the other way around.